### PR TITLE
Proposal page view

### PR DESCRIPTION
**Pull request by Mati0x**
#240

### CODE REVIEW

File: apps/web/app/(app)/gardens/[chain]/[garden]/pool/[poolId]/proposals/[proposalId]/page.tsx [hunk 1]
```diff
@@ -215,71 +219,65 @@ export default async function Proposal({
...
           </div>
+          <p>{description}</p>
+          <div className="flex flex-col gap-2">
+            {!isSignalingType && (
+              <>
+                <Statistic
+                  label={"requested amount"}
+                  icon={<InformationCircleIcon />}
+                >
+                  <DisplayNumber
```
Comment (Lucianosc) [inline apps/web/app/(app)/gardens/[chain]/[garden]/pool/[poolId]/proposals/[proposalId]/page.tsx:255]: DisplayNumber already adds up a ml-2 maybe we should get rid of it (or at least consider it in Statistic)

File: apps/web/app/(app)/gardens/[chain]/[garden]/pool/[poolId]/proposals/[proposalId]/page.tsx [hunk 2]
```diff
@@ -215,71 +219,65 @@ export default async function Proposal({
...
+                    tokenSymbol={tokenSymbol}
+                    compact={true}
+                    className="font-bold text-black"
+                  />
+                </Statistic>
+                <Statistic label={"beneficiary"} icon={<UserIcon />}>
+                  <EthAddress
+                    address={beneficiary}
+                    actions="copy"
+                    icon="identicon"
```
Comment (Lucianosc) [inline apps/web/app/(app)/gardens/[chain]/[garden]/pool/[poolId]/proposals/[proposalId]/page.tsx:266]: the identicon looks awful :joy:

**Review by Lucianosc — CHANGES_REQUESTED** [review]

File: apps/web/app/(app)/gardens/[chain]/[garden]/pool/[poolId]/proposals/[proposalId]/page.tsx (hunk 2 above)
Comment (Mati0x) [reply apps/web/app/(app)/gardens/[chain]/[garden]/pool/[poolId]/proposals/[proposalId]/page.tsx:266]: I agree it not look  very nice, I can remove for now yeahp

File: apps/web/app/(app)/gardens/[chain]/[garden]/pool/[poolId]/page.tsx
Comment (Corantin) [inline apps/web/app/(app)/gardens/[chain]/[garden]/pool/[poolId]/page.tsx:1]: I have the following problem in the pool view in the console: 
image

Comment (Corantin) [reply apps/web/app/(app)/gardens/[chain]/[garden]/pool/[poolId]/page.tsx:1]: Dont know if this code is telling you something ? 
image

Comment (Corantin) [reply apps/web/app/(app)/gardens/[chain]/[garden]/pool/[poolId]/page.tsx:1]: Happened in that uri: https://gardens-v2-git-proposal-page-view-1hive.vercel.app/gardens/421614/0xcc6c8b9f745db2277f7aac1bc026d5c2ea7bd88d/pool/356

Comment (Corantin) [reply apps/web/app/(app)/gardens/[chain]/[garden]/pool/[poolId]/page.tsx:1]: Step to repro:
1. Go to this page:  https://gardens-v2-git-proposal-page-view-1hive.vercel.app/gardens/421614/0xcc6c8b9f745db2277f7aac1bc026d5c2ea7bd88d/pool/356
2. Scroll down and up
3. Eventually I have the problem described here

Comment (Corantin) [reply apps/web/app/(app)/gardens/[chain]/[garden]/pool/[poolId]/page.tsx:1]: Happened only once

File: apps/web/app/(app)/gardens/[chain]/[garden]/pool/[poolId]/page.tsx [hunk 3]
```diff
@@ -60,101 +72,89 @@ export default async function Pool({
...
               />
-            </>
+            </Statistic>
+          )}
+
+          <Statistic
+            label="voting weight system"
+            icon={<InformationCircleIcon />}
+          >
+            <div className="flex flex-col gap-3 sm:flex-row">
```
Comment (Corantin) [inline apps/web/app/(app)/gardens/[chain]/[garden]/pool/[poolId]/page.tsx:106]: Just making sure we both understand what is done here with the sm:
You wanted to above mobile resolution right?
SM = resolution >= 640px and here is how we target mobile devices:
image

Comment (Corantin) [reply apps/web/app/(app)/gardens/[chain]/[garden]/pool/[poolId]/page.tsx:106]: Source: https://tailwindcss.com/docs/responsive-design

File: apps/web/app/(app)/gardens/[chain]/[garden]/pool/[poolId]/proposals/[proposalId]/page.tsx [hunk 4]
```diff
@@ -10,7 +10,11 @@ import {
   getProposalDataQuery,
 } from "#/subgraph/.graphclient";
 import { formatTokenAmount, calculatePercentageBigInt } from "@/utils/numbers";
+import Image from "next/image";
 import { getIpfsMetadata } from "@/utils/ipfsUtils";
+import { UserIcon, InformationCircleIcon } from "@heroicons/react/24/outline";
+import { proposalStatus, poolTypes } from "@/types";
+import { proposalImg } from "@/assets";
```
Comment (Corantin) [inline apps/web/app/(app)/gardens/[chain]/[garden]/pool/[poolId]/proposals/[proposalId]/page.tsx:17]: Looks like this image is too much pixelated, maybe we should implement the proposal identicon very soon, I will add an issue for that I the board:
https://github.com/orgs/1Hive/projects/12/views/1?pane=issue&itemId=69082828

File: apps/web/app/(app)/gardens/[chain]/[garden]/pool/[poolId]/proposals/[proposalId]/page.tsx [hunk 5]
```diff
@@ -90,7 +94,7 @@ export default async function Proposal({
   const stakedAmount = proposalData.stakedAmount as bigint;
   const metadata = proposalData.metadata;
 
-  const isSignalingType = proposalType == 0;
+  const isSignalingType = poolTypes[proposalType] == "signaling";
```
Comment (Corantin) [inline apps/web/app/(app)/gardens/[chain]/[garden]/pool/[poolId]/proposals/[proposalId]/page.tsx:97]: <3

File: apps/web/app/(app)/gardens/[chain]/[garden]/pool/[poolId]/proposals/[proposalId]/page.tsx [hunk 6]
```diff
@@ -215,71 +219,61 @@ export default async function Proposal({
...
+            <div className="mb-4 flex flex-col items-start gap-4 sm:mb-2 sm:flex-row sm:items-center sm:justify-between sm:gap-2">
+              <h2>
+                Proposal #{proposalIdNumber} - {title}
+              </h2>
+              <Badge type={proposalType} />
+            </div>
+            <div className="flex items-center justify-between gap-4 sm:justify-start">
+              <Badge status={status} />
+              <p className="font-semibold">
+                {prettyTimestamp(proposalData?.createdAt || 0)}
```
Comment (Corantin) [inline apps/web/app/(app)/gardens/[chain]/[garden]/pool/[poolId]/proposals/[proposalId]/page.tsx:243]:
```suggestion
                {prettyTimestamp(proposalData?.createdAt ?? 0)}
```
Nothing will change but its would just save an extra operation in the case createdAt is 0

Comment (Corantin) [reply apps/web/app/(app)/gardens/[chain]/[garden]/pool/[poolId]/proposals/[proposalId]/page.tsx:243]: ?? : Null ord Undefined will coalesce (use right part)
|| : All falsy values including null or undefined but also 0 and ""

File: apps/web/assets/index.ts [hunk 7]
```diff
@@ -35,5 +36,6 @@ export const grassBrown = grassBrown_;
 export const walletIcon = walletIcon_;
 export const poolGrassBlue = poolGrassBlue_;
 export const commImg = commImg_;
+export const proposalImg = proposalImg_;
```
Comment (Corantin) [inline apps/web/assets/index.ts:39]: We we dont use direct export ? 
would be:
> export { default } from "./ProposalImage.png";

File: apps/web/components/Badge.tsx [hunk 8]
```diff
@@ -12,6 +12,7 @@ type BadgeProps = {
   label?: string;
   classNames?: string;
   icon?: React.ReactNode;
+  isCapitalize?: boolean;
```
Comment (Corantin) [inline apps/web/components/Badge.tsx:15]:
```suggestion
  capitalized?: boolean;
```

Comment (Corantin) [reply apps/web/components/Badge.tsx:15]: You free to ignore that

**Review by Corantin — CHANGES_REQUESTED** [review]

File: apps/web/app/(app)/gardens/[chain]/[garden]/pool/[poolId]/page.tsx
Comment (Mati0x) [reply apps/web/app/(app)/gardens/[chain]/[garden]/pool/[poolId]/page.tsx:1]: No worries here I think, I did have have that issue and also dont have that error in my console

File: apps/web/app/(app)/gardens/[chain]/[garden]/pool/[poolId]/page.tsx (hunk 3 above)
Comment (Mati0x) [reply apps/web/app/(app)/gardens/[chain]/[garden]/pool/[poolId]/page.tsx:106]: Yeah, so we apply classes for mobile and in the case we wanna have different styles above that resolution we do sm: ... , or md: ...
For the moment we just add some sm: ... styles, so we have at least components and page details in a nice order and stacked

File: apps/web/app/(app)/gardens/[chain]/[garden]/pool/[poolId]/proposals/[proposalId]/page.tsx (hunk 4 above)
Comment (Mati0x) [reply apps/web/app/(app)/gardens/[chain]/[garden]/pool/[poolId]/proposals/[proposalId]/page.tsx:17]: Agreed yesss ... I propose to add this issue in #257 where we are gonna upgrade the `Proposal` component and by consequence we can update the proposal page also

File: apps/web/assets/index.ts (hunk 7 above)
Comment (Mati0x) [reply apps/web/assets/index.ts:39]: idk why we have this structure to export images and svgs really

Comment (Corantin) [reply apps/web/assets/index.ts:39]: Yeah I think there is better way but we can tackle it later.

**Review by Corantin — APPROVED** [review]

File: apps/web/app/(app)/gardens/[chain]/[garden]/pool/[poolId]/proposals/[proposalId]/page.tsx (hunk 6 above)
Comment (Corantin) [reply apps/web/app/(app)/gardens/[chain]/[garden]/pool/[poolId]/proposals/[proposalId]/page.tsx]: Need reason to resolve this :p

**Review by Lucianosc — APPROVED** [review]

**Pull request merged**